core/server/master/src/main/java/alluxio/master/file/meta/SyncCheck.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Returned by {@link UfsSyncPathCache#shouldSyncPath} indicating whether
+ * a sync is needed as well as the time of the last sync if one is not needed.
+ * After the sync is performed, calling {@link SyncCheck#syncSuccess()} or
+ * {@link SyncCheck#skippedSync()} will result in creating an object containing
+ * information pertaining to the result of the sync.
+ */
+public class SyncCheck {
+  private final boolean mShouldSync;
+  private final long mLastSyncTime;
+
+  /** Sync is not needed, with no last sync time. **/
+  public static final SyncCheck SHOULD_NOT_SYNC = new SyncCheck(
+      false, 0);
+  /** Sync is needed. **/
+  public static final SyncCheck SHOULD_SYNC = new SyncCheck(
```

Comment:
These constants should not be public: they may be mistaken as enum variants and people can be lured into comparing them by doing `if (mySyncCheckObj == SyncCheck.SHOULD_NOT_SYNC)` while the correct way is `if (!mySyncCheckObj.isShouldSync())`. 

I'd suggest to hide these constants behind two public static factory methods `SyncCheck.shouldSync()` and `SyncCheck.shouldNotSync()`.